logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -180,6 +180,16 @@ def each(path = nil, &block)
     end
     alias_method :all, :each
 
+    def prune(path)
+      key_paths = path.gsub(/^#{KEY_PATH_SEPARATOR}+/, "").split(KEY_PATH_SEPARATOR)
+      key_paths = key_paths.map {|k| k.to_sym }
+      @structured_lookup_mutex.synchronize do
+        keys_to_delete = @fast_lookup.keys.select {|namespace, _| (key_paths - namespace).empty? }
+        keys_to_delete.each {|k| @fast_lookup.delete(k) }
```

Comment:
true nevermind, this is was happen when we mixed java ruby and concurrency, we can't always have nice things.